math/average.c

```diff
@@ -0,0 +1,25 @@
+#include <stdio.h>
+#include <assert.h>
+
+//average of an array
+
+double average(int avg[]) {
+    double result = 0;
+    int size=0;
+    
+    while(avg[size+1]!='\0'){
+        size++;
+    }
+
+    for (int i = 0; i < size; i++) {
+        result += avg[i];
+    }
+
+    return result / size;
+}
+
+int main() {
+    int test[] = {1,2,3,4,5};
```

Comment:
You could add a few different tests to ensure the robustness, so for example using all negative numbers, a mix of negative and positive, you could go for large numbers close to the maximum size of an `int` and see what happens there (most likely an overflow and the wrong result) which is what you want to see from tests anyway. Play about with testing. It's great at highlighting the faults.